src/darwin/Framework/CHIP/MTRMetricsCollector.mm

```diff
@@ -41,60 +46,50 @@ - (instancetype)initWithMetricEvent:(const MetricEvent &)event
         return nil;
     }
 
+    _type = event.type();
+
+    using EventType = MetricEvent::Type;
+    switch (_type) {
+    // Capture timepoint for begin and end to calculate duration
+    case EventType::kBeginEvent:
+    case EventType::kEndEvent:
+        _timePoint = chip::System::SystemClock().GetMonotonicMicroseconds64();
+        break;
+    case EventType::kInstantEvent:
+        _timePoint = chip::System::Clock::Microseconds64(0);
+        break;
+    }
+
     using ValueType = MetricEvent::Value::Type;
     switch (event.ValueType()) {
     case ValueType::kInt32:
         _value = [NSNumber numberWithInteger:event.ValueInt32()];
         break;
     case ValueType::kUInt32:
-        _value = [NSNumber numberWithInteger:event.ValueUInt32()];
+        _value = [NSNumber numberWithUnsignedInteger:event.ValueUInt32()];
         break;
     case ValueType::kChipErrorCode:
-        _value = [NSNumber numberWithInteger:event.ValueErrorCode()];
+        _errorCode = [NSNumber numberWithUnsignedInteger:event.ValueErrorCode()];
         break;
     case ValueType::kUndefined:
-    default:
-        _value = nil;
+        break;
     }
-
-    _timePoint = chip::System::SystemClock().GetMonotonicMicroseconds64();
-    _duration = chip::System::Clock::Microseconds64(0);
     return self;
 }
 
-- (void)setDurationFromMetricData:(MTRMetricsData *)fromData
-{
-    _duration = _timePoint - fromData->_timePoint;
-}
-
-- (NSNumber *)timePointMicroseconds
+- (void)setDurationFromMetricDataAndClearCounters:(MTRMetricData *)fromData
 {
-    return [NSNumber numberWithUnsignedLongLong:_timePoint.count()];
-}
+    auto duration = _timePoint - fromData->_timePoint;
+    _durationMicroseconds = [NSNumber numberWithUnsignedLongLong:duration.count()];
 
-- (NSNumber *)durationMicroseconds
-{
-    return [NSNumber numberWithUnsignedLongLong:_duration.count()];
+    // Clear timepoints to minimize history
+    _timePoint = fromData->_timePoint = chip::System::Clock::Microseconds64(0);
 }
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"MTRMetricsData: Value = %@, TimePoint = %@, Duration = %@ us", self.value, self.timePointMicroseconds, self.durationMicroseconds];
-}
-
-- (NSDictionary *)toDictionary
-{
-    NSMutableDictionary * dictRepresentation = [NSMutableDictionary dictionary];
-    if (self.value) {
-        [dictRepresentation setValue:self.value forKey:kMTRMetricDataValueKey];
-    }
-    if (auto tmPt = self.timePointMicroseconds) {
-        [dictRepresentation setValue:tmPt forKey:kMTRMetricDataTimepointKey];
-    }
-    if (auto duration = self.durationMicroseconds) {
-        [dictRepresentation setValue:duration forKey:kMTRMetricDataDurationKey];
-    }
-    return dictRepresentation;
+    return [NSString stringWithFormat:@"<MTRMetricData>: Type %d, Value = %@, Error Code = %@, Duration = %@ us",
```

Comment:
My understanding is that the closing `>` should be after the member details, no?  So `<MTRMetricData: Type %d, Value = %@, Error Code = %@, Duration = %@ us>`

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/32439